File: js/SHA1/main.js

const SHA1 = require('./SHA1');

function main() {
    const data = "abc";
    console.log("Message: " + data);

    const sha1 = new SHA1();
    sha1.update(data);
    const digest = sha1.final();

    let hex = "";
    for (let i = 0; i < digest.length; i++) {
        let h = digest[i].toString(16);
        if (h.length < 2) h = '0' + h;
        hex += h;
    }
    
    console.log("SHA1: " + hex);

    const expected = "a9993e364706816aba3e25717850c26c9cd0d89d";
    console.log("Expect: " + expected);
}

main();

File: c/SHA1/main.c

#include <stdio.h>
#include <string.h>
#include "SHA1.h"

int main()
{
    uint8_t digest[20];
    const char *data = "abc";
    
    printf("Message: %s\n", data);
    
    SHA1((const uint8_t*)data, strlen(data), digest);
    
    printf("SHA1: ");
    for (int i = 0; i < 20; i++) {
        printf("%02x", digest[i]);
    }
    printf("\n");
    
    const char *expected = "a9993e364706816aba3e25717850c26c9cd0d89d";
    printf("Expect: %s\n", expected);
    
    return 0;
}
